params.go

```diff
@@ -18,6 +18,7 @@ const (
 type Params struct {
 	Exp            []string
 	Meta           map[string]string
+	Extra          url.Values
```

Comment:
See my comment in the PR description for why we don't want a `map[string]interface{}`. A `map[string]string` doesn't support an array of multiple values for the same key which is a valid form. `url.Values` seems like it matches the semantics that we're using this for.